File: Python/Search/binary_search.py
def binary_search(arr, target):
    arr.sort()

    head = 0
    tail = len(arr) - 1

    while tail >= head:
        mid = (head + tail) // 2
        if arr[mid] == target:
            return str(mid) + ' is in there'
        elif arr[mid] > target:
            tail = mid - 1
        elif arr[mid] < target:
            head = mid + 1

    return 'Not found'


if __name__ == '__main__':
    lst = [1, 2, 4, 5, 6, 7, 2, 3, 5, 6, 7, 8]
    print(binary_search(lst, 4))

File: JavaScript/Search/BinarySearch.js
/**
 * Created by ys on 18. 4. 16.
 */
((array, target) => {
    "use strict"

    array.sort((a, b) => {
        return a - b
    })

    let head = 0
    let tail = array.length - 1

    while (tail >= head) {
        let mid = Math.floor((head + tail) / 2)
        if (array[mid] === target) {
            console.log(`${mid} is in there`)
            return mid
        } else if (array[mid] > target) {
            tail = mid - 1
        } else if (array[mid] < target) {
            head = mid + 1
        }
    }

    console.log('Not found')
    return -1

})([1,2,4,5,6,7,2,3,5,6,7,8], 4)

